includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -68,11 +81,12 @@ public function init(): string {
 			'WCPAY_PRODUCT_DETAILS',
 			'wcpayStripeSiteMessaging',
 			[
-				'price'          => WC_Payments_Utils::prepare_amount( $price, $currency_code ),
-				'currency'       => $currency_code,
-				'country'        => $billing_country,
-				'publishableKey' => $this->account->get_publishable_key( WC_Payments::mode()->is_test() ),
-				'paymentMethods' => array_values( $bnpl_payment_methods ),
+				'price'               => WC_Payments_Utils::prepare_amount( $price, $currency_code ),
+				'currency'            => $currency_code,
+				'variationsPriceList' => $variations_price_list,
```

Comment:
Looks better, I'd go into step further and send base product + variation as array always.

Reply:
Thanks! I have included the base product in the `variationsPriceList` array . I had to use an associative array, since the `show_event` provides the variation ID and i use that to pull the required price. For the base product, i am using an index `base_product`. Hope that is okay.

Reply:
Looks like the right direction. I believe sending 'price' and 'currency' is excessive now?

Reply:
`currency` is a part of the original code written by Miguel. It is needed to initialize the [messaging element here](https://github.com/Automattic/woocommerce-payments/blob/e4514075fba7e9366006d4a9638922fa823c9c8c/client/product-details/bnpl-site-messaging/index.js#L15).